library/datetime/deconstruct_keys_spec.rb

```diff
@@ -0,0 +1,33 @@
+require_relative '../../spec_helper'
+require 'date'
+
+ruby_version_is "3.2" do
+  describe "DateTime#deconstruct_keys" do
+    it "returns whole hash for nil as an argument" do
+      d = DateTime.new(2022, 10, 5, 13, 30)
+      res = { year: 2022, month: 10, day: 5, yday: 278, wday: 3, hour: 13,
+              min: 30, sec: 0, sec_fraction: (0/1), zone: "+00:00" }
+      d.deconstruct_keys(nil).should == res
+    end
+
+    it "returns only specified keys" do
+      d = DateTime.new(2022, 10, 5, 13, 39)
+      d.deconstruct_keys([:zone, :hour]).should == { zone: "+00:00", hour: 13 }
+    end
+
+    it "requires one argument" do
+      -> {
+        DateTime.new(2022, 10, 5, 13, 30).deconstruct_keys
+      }.should raise_error(ArgumentError)
+    end
+
+    it "accepts only array or nil as an argument" do
+      d = DateTime.new(2022, 10, 5, 13, 30)
+
+      -> { d.deconstruct_keys(1) }.should raise_error(TypeError)
+      -> { d.deconstruct_keys("asd") }.should raise_error(TypeError)
+      -> { d.deconstruct_keys(:x) }.should raise_error(TypeError)
+      -> { d.deconstruct_keys({}) }.should raise_error(TypeError)
+    end
+  end
+end
```

Comment:
*suggestion:* I would consider adding some other cases similar to specs for `Struct#deconstruct_keys`, e.g.

- when passed String, not Symbol - `DateTime.now.deconstruct_keys(['year']) # => {}`
- when passed empty Array - `DateTime.now.deconstruct_keys([]) # => {}`
- when passed not existing key - `DateTime.now.deconstruct_keys([:year, :a]) # => {:year=>2023}`